File: frontend/Moment/app/src/main/java/snu/swpp/moment/ui/main_writeview/MomentUiState.java
package snu.swpp.moment.ui.main_writeview;

import java.util.ArrayList;
import snu.swpp.moment.data.model.MomentPairModel;

public class MomentUiState {

    private final int error;
    private final ArrayList<MomentPairModel> momentPairsList;

    public MomentUiState(int error, ArrayList<MomentPairModel> list) {
        this.error = error;
        this.momentPairsList = list;
    }

    public int getError() {
        return error;
    }

    public ArrayList<MomentPairModel> getMomentPairsList() {
        return momentPairsList;
    }

    public int getMomentPairsListSize() {
        return momentPairsList.size();
    }

    public void addMomentPair(MomentPairModel momentPair) {
        momentPairsList.add(momentPair);
    }

}

File: frontend/Moment/app/src/main/java/snu/swpp/moment/ui/main_writeview/uistate/MomentUiState.kt
package snu.swpp.moment.ui.main_writeview.uistate

import snu.swpp.moment.data.model.MomentPairModel

class MomentUiState(
    val error: Exception?,
    val momentPairList: List<MomentPairModel>,
) {
    companion object {
        @JvmStatic
        fun withError(error: Exception): MomentUiState {
            return MomentUiState(error, emptyList())
        }
    }
    
    fun getNumMoments(): Int {
        return momentPairList.size
    }
}